includes/Data/Patterns.php

```diff
@@ -213,16 +226,20 @@ public static function get_pattern_from_slug( $pattern_slug ) {
 	/**
 	 * Replace the header menu slug in the patterns array
 	 *
-	 * @param array  $patterns Patterns for the specific step
-	 * @param string $header_menu_slug header menu slug choosen by the user
+	 * @param array $patterns Patterns for the specific step
 	 *
 	 * @return array
 	 */
-	private static function replace_header_menu_slug( $patterns, $header_menu_slug ) {
-		foreach ( $patterns as $slug => $slug_details ) {
-			if ( isset( $slug_details['replace'] ) && true === $slug_details['replace'] ) {
-				unset( $patterns[ $slug ] );
-				$patterns = array_merge( array( $header_menu_slug => $slug_details ), $patterns );
+	private static function replace_header_menu_slug( $patterns ) {
+		// fetch the selected header menu slug from DB
+		$flow_data        = \get_option( Options::get_option_name( 'flow' ) );
+		$header_menu_slug = explode( '/', $flow_data['data']['partHeader'] )[1];
+		if ( ! empty( $header_menu_slug ) ) {
+			foreach ( $patterns as $slug => $slug_details ) {
+				if ( isset( $slug_details['replace'] ) && true === $slug_details['replace'] ) {
```

Comment:
Yeah, since the header menu slug name has to be changed in the patterns list before the block grammar for each pattern is generated. Iterating through the list and checking for the `replace` key would not serve the purpose of changing the header menu slug.